gobblin-metastore/src/main/java/gobblin/metastore/StateStore.java

```diff
@@ -127,6 +127,18 @@ public T get(String storeName, String tableName, String stateId)
       throws IOException;
 
   /**
+   * Get a {@link State} with a given state ID from a the current table.
+   *
+   * @param storeName store name
+   * @param stateId state ID
+   * @return {@link State} with the given state ID or <em>null</em>
+   *         if the state with the given state ID does not exist
+   * @throws IOException
+   */
+  public T getCurrent(String storeName, String stateId)
```

Comment:
If I remember correctly, the reason for the `getCurrent` vs `getAllCurrent` is that it is mirroring the `get` and `getAll` methods which took table name.

DatasetStateStore has existed for more than a year and this change was compatible with how it worked.  As part of rebasing this changed onto the latest master, I made the new implementations of state store work without a `current` state.